gobblin-core/src/main/java/gobblin/metrics/GobblinMetrics.java

```diff
@@ -461,15 +472,35 @@ private void buildKafkaMetricReporter(Properties properties) {
       return;
     }
 
-    KafkaReporter.Builder<?> builder = formatEnum.reporterBuilder(this.metricContext);
+    if(topicMetrics.or(topicDefault).isPresent()) {
```

Comment:
MetricReporter is part of gobblin-metrics, not gobblin-core, so it is not aware of any gobblin runtime configurations.